content/en/docs/ops/best-practices/observability/index.md

```diff
@@ -0,0 +1,268 @@
+---
+title: Observability Best Practices
+description: Best practices for observing applications using Istio.
+force_inline_toc: true
+weight: 50
+---
+
+## Using Prometheus for production-scale monitoring
+
+The recommended approach for production-scale monitoring of Istio meshes with Prometheus
+is to use [hierarchical federation](https://prometheus.io/docs/prometheus/latest/federation/#hierarchical-federation)
+in combination with a collection of [recording rules](https://prometheus.io/docs/prometheus/latest/configuration/recording_rules/).
+
+In default deployments of Istio, a deployment of [Prometheus](http://prometheus.io) is
+provided for collecting metrics generated for all mesh traffic. This deployment of
+Prometheus is intentionally deployed with a very short retention window (6 hours). The
+default Prometheus deployment is also configured to collect metrics from each Envoy proxy
+running in the mesh, augmenting each metric with a set of labels about their origin (`instance`,
+`pod_name`, and `namespace`).
+
+While the default configuration is well-suited for small clusters and monitoring for short time horizons,
+it is not suitable for large-scale meshes or monitoring over a period of days or weeks. In particular,
+the introduced labels can increase metrics cardinality, requiring a large amount of storage. And, when trying
+to identify trends and differences in traffic over time, access to historical data can be paramount.
+
+{{< image width="80%"
+    link="./production-prometheus.svg"
+    alt="Architecture for production monitoring of Istio using Prometheus."
+    caption="Production-scale Istio monitoring with Istio"
+    >}}
+
+### Workload-level aggregation via recording rules
+
+In order to aggregate metrics across instances and pods, update the default Prometheus configuration with
+the following recording rules:
+
+{{< text yaml >}}
+groups:
+- name: "istio.recording-rules"
+  interval: 30s
```

Comment:
So interestingly we discovered that having this at 30s leads to data missing in your graphs (despite envoy being scraped at 30s, and these rules being evaluated every 30s).  You need the recording rules to evaluate more frequently than the scrape.

The sweet spot for us which gives us nice fidelity in the graphs is:

```
global:
  scrape_interval: 15s
  evaluation_interval: 15s

...

groups:
- name: "istio.recording-rules-fast"
  interval: 5s
  rules:
  - any sum(irate(....

- name: "istio.recording-rules-fast"
  interval: 5s
  rules:
  - any histogram_quantile(...
```

Reply:
Ok. I've updated the examples to use `5s`. Thanks for the info.